server.js

```diff
@@ -20,4 +21,46 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
+app.get("/messages/search", (req, res) => {
+  const { term } = req.query;
+  console.log(term);
+  const filterMessages = messages.filter((message) =>
+    message.text.toLowerCase().includes(term.toLowerCase())
+  );
+  console.log(filterMessages);
+  res.send(filterMessages);
+});
+
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+app.get("/messages/:id", function (req, res) {
+  const id = req.params.id;
+  messages = messages.filter((message) => message.id === Number(id));
```

Comment:
To convert to an integer, use `parseInt` rather than the `Number` constructor - in this case it'd work out the same, but we tend to prefer `parseInt` as you can pass in a second parameter to specify the numeral system to parse the string as (e.g. hexidecimal, decimal, oct, etc)